src/buza/models.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import unicode_literals
+
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.contrib.auth.models import PermissionsMixin
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from project.boards.models import Board
+
+from .managers import UserManager
+
+
+# Models will be added to the db
+
+
+class BuzaUser(AbstractBaseUser, PermissionsMixin):
+    # authentication feilds
+    email = models.EmailField(
+        _('email address'),
+        unique=True,
+        blank=True,
+    )
+    phone = models.CharField(
+        _('phone number'),
+        blank=True,
+        unique=True,
+        max_length=11,
+    )
+    user_name = models.CharField(
+        _('user name'),
+        blank=True,
+        null=True,
+        unique=True,
+        max_length=15,
+    )
+    # school fields
+    school = models.CharField(_('school name'), blank=True, null=True, max_length=100)
+    school_address = models.CharField(
+        _('school address'),
+        blank=True, null=True,
+        max_length=300,
+    )
+    grade = models.IntegerField(blank=True, default=7)
+    # users personal feilds
+    photo = models.ImageField(_('profile photo'),
+                              upload_to='users/%Y/%m/%d',
+                              blank=True)
+    bio = models.CharField(blank=True, null=True, max_length=250)
+    date_joined = models.DateTimeField(_('date joined'), auto_now_add=True)
+    is_active = models.BooleanField(_('active'), default=True)
+    full_name = models.CharField(
+        _('name and surname'),
+        blank=True, null=True,
+        max_length=100,
+    )
+
+    # users can have multiple boards
+    subjects = models.ManyToManyField(Board, related_name="my_boards")
```

Comment:
Should we update this terminology to "subjects" for consistency, rather than "boards"?